### PR TITLE
[ts-sdk] Add `signMessage` and `signTransaction` APIs

### PR DESCRIPTION
**Pull request by Jordan-Mysten**
The SDK does not currently support `signMessage` and `signTransaction`, so I added this missing APIs. I reworked how we construct intents and signed messages to clean up some duplication. I also changed the return types from these APIs to make them return the serialized signature string. The motivation for strings is that we're preparing something to be submitted over an API, and it ends up getting string-encoded at a later point anyway (often in the next function the data is passed to), so we might as well just make the return more sensible. This also lets these APIs be used over JSON-bridges without additional work (e.g. no need to do additional work to serialize / parse the data). Strings are also not mutable like Uint8Array which also should help denote that these things are finalized and ready for submission to the network.

### CODE REVIEW

**Review by mamos-mysten — APPROVED** [review]
code lgtm

File: sdk/typescript/src/utils/intent.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Mysten Labs, Inc.
```
Comment (joyqvq) [inline sdk/typescript/src/utils/intent.ts:1]: 💯

File: sdk/typescript/src/signers/signer-with-provider.ts [hunk 2]
```diff
@@ -89,40 +89,64 @@ export abstract class SignerWithProvider implements Signer {
...
    */
-  async signAndExecuteTransaction(
+  async signMessage(message: Uint8Array): Promise<SignedMessage> {
+    const messageBytes = messageWithIntent(
+      IntentScope.PersonalMessage,
+      message,
+    );
+    const signature = await this.signData(messageBytes);
+    return {
+      messageBytes: toB64(messageBytes),
```
Comment (joyqvq) [inline sdk/typescript/src/signers/signer-with-provider.ts:101]: dont think we need to care about the message bytes to return. the verifier should know what intent they are verifying to and construct this themselves. (to save communication cost too)

File: sdk/typescript/src/signers/signer.ts [hunk 3]
```diff
@@ -11,8 +11,11 @@ import { PublicKey, SignatureScheme } from '../cryptography/publickey';
  */
 export type SignaturePubkeyPair = {
   signatureScheme: SignatureScheme;
-  signature: Uint8Array;
-  pubKey: PublicKey;
+  /** Base64-encoded signature */
+  signature: string;
+  /** Base64-encoded public key */
+  pubKey: string;
+  serializedSignature: string;
```
Comment (joyqvq) [inline sdk/typescript/src/signers/signer.ts:18]: is this just for convenience here? serializeSignature should already contain (signatureScheme, signature, pubKey)

File: sdk/typescript/src/signers/signer-with-provider.ts (hunk 2 above)
Comment (Jordan-Mysten) [reply sdk/typescript/src/signers/signer-with-provider.ts]: Good call, removed and added `verifyMessage`.

File: sdk/typescript/src/signers/signer.ts (hunk 3 above)
Comment (Jordan-Mysten) [reply sdk/typescript/src/signers/signer.ts:18]: Yeah this is just convenience. This allows this signature + data to directly be submitted without another serialization step later, while simultaneously allowing the individual fields to be used as desired (and presented in a JSON-friendly format that we can send between different contexts like wallet / dapps).

Comment (Jordan-Mysten) [reply sdk/typescript/src/signers/signer.ts:18]: The alternative is just to return the `serializedSignature`, and then extract everything later on, it just means we need to convert data quite a bit which might be annoying. But I'm open to suggestions here if you think it'd be more worthwhile to just return that and then have utils to extract it out.

**Review by joyqvq — APPROVED** [review]
lgtm!

File: sdk/typescript/src/signers/signer.ts (hunk 3 above)
Comment (Jordan-Mysten) [reply sdk/typescript/src/signers/signer.ts]: I ended up just using the serialized signature and adding new `toSerializedSignature` and `fromSerializedSignature` utils to help with this!

**Pull request merged**